provider/spark.go

```diff
@@ -590,20 +605,46 @@ func (q defaultPythonOfflineQueries) materializationCreate(schema ResourceSchema
 			SELECT
 				ord.entity 
 				,ord.value 
-				--,ord.ts -- TODO: determine if we even need to add this zeroed-out timestamp column
+				,ord.ts
 			FROM
 				max_row_per_entity maxr
 			JOIN ordered_rows ord
 				ON ord.entity = maxr.entity AND ord.row_number = maxr.max_row
 			ORDER BY
 				maxr.max_row DESC`, schema.Entity, schema.Value, schema.Entity)
 	}
-	return fmt.Sprintf(
-		"SELECT entity, value, ts, ROW_NUMBER() over (ORDER BY (SELECT NULL)) AS row_number, rn2 FROM "+
-			"(SELECT entity, value, ts, ROW_NUMBER() OVER (PARTITION BY entity ORDER BY ts DESC) AS rn2 FROM "+
-			"(SELECT entity, value, ts, rn FROM (SELECT %s AS entity, %s AS value, %s AS ts, "+
-			"ROW_NUMBER() OVER (ORDER BY (SELECT NULL)) AS rn FROM %s) t ORDER BY rn DESC) t2 ) t3 WHERE rn2=1",
-		schema.Entity, schema.Value, timestampColumn, "source_0")
+	return fmt.Sprintf(`SELECT
```

Comment:
did the formatting bork out?

Reply:
yeahh i prefer the previous formatting lol; or maybe have a file for sql string constants

Reply:
I'd prefer a separate SQL file. Inlined SQL is often hard to read for subqueries, etc. I'll give this a shot.